src/test/ui/pattern/usefulness/issue-78549-ref-pat-and-str.rs

```diff
@@ -0,0 +1,14 @@
+// check-pass
+// From https://github.com/rust-lang/rust/issues/78549
+
+fn main() {
+    match "foo" {
```

Comment:
Could you add one more check where `&_` occurs in a nested pattern, just to make sure? E.g.
```rust
match (&"foo", "bar") {
    (&"foo", &_) => {},
    (&&_, &_) => {},
}
```

Reply:
Oh, well caught, that indeed broke my code. That gave me a less error-prone way of solving the issue.